tests/test_frictionless_to_ckan.py

```diff
@@ -1,30 +1,305 @@
 # coding=utf-8
 
-import pytest
+import frictionless_ckan_mapper.frictionless_to_ckan as converter
 
-import frictionless_ckan_mapper.frictionless_to_ckan as frictionless_to_ckan
 
+class TestResourceConversion:
+    def test_extras(self):
+        indict = {
+            'extras': {
+                'title_cn': u'國內生產總值',
+                'years': [2015, 2016],
+                'last_year': 2016,
+                'location': {'country': 'China'}
+            }
+        }
+        out = converter.resource(indict)
+        exp = {
+            'extras': [
+                {'key': 'last_year', 'value': 2016},
+                {'key': 'location', 'value': '{"country": "China"}'},
+                {'key': 'title_cn', 'value': u'國內生產總值'},
+                {'key': 'years', 'value': '[2015, 2016]'}
+            ]
+        }
+        out['extras'] = sorted(out['extras'], key = lambda i: i['key'])
+        assert out['extras'] == exp['extras']
 
-converter = frictionless_to_ckan.FrictionlessToCKAN()
+    def test_name_is_used_if_theres_no_title(self):
+        indict = {'name': 'gdp'}
+        out = converter.resource(indict)
+        assert out.get('name') == indict['name']
 
+    def test_resource_title_is_used_as_name(self):
+        indict = {
+            'name': 'gdp',
+            'title': 'Gross domestic product',
+        }
 
-class TestResourceConversion:
-    pass
+        out = converter.resource(indict)
+        assert out.get('name') == indict['title']
+
+    def test_path_to_url(self):
+        # Test remote path
+        indict = {'path': 'http://www.somewhere.com/data.csv'}
```

Comment:
please lay out like:

```
indict = {
     ...
}
```

it's easier to read and more consistent if we add other values.